diff --git a/android/src/main/java/dev/matinzd/healthconnect/records/ReactExerciseSessionRecord.kt b/android/src/main/java/dev/matinzd/healthconnect/records/ReactExerciseSessionRecord.kt
index 3dfdee5ef7dd49c6285fa9666ee57ee28d9595eb..ad18a804f603201f623fcb37a261898cd96444e5 100644
--- a/android/src/main/java/dev/matinzd/healthconnect/records/ReactExerciseSessionRecord.kt
+++ b/android/src/main/java/dev/matinzd/healthconnect/records/ReactExerciseSessionRecord.kt
@@ -45,14 +45,14 @@ class ReactExerciseSessionRecord : ReactHealthRecordImpl<ExerciseSessionRecord>
         ),
         notes = it.getString("notes"),
         title = it.getString("title"),
-        laps = it.getArray("samples")?.toMapList()?.map { sample ->
+        laps = it.getArray("laps")?.toMapList()?.map { sample ->
           ExerciseLap(
             startTime = Instant.parse(sample.getString("startTime")),
             endTime = Instant.parse(sample.getString("endTime")),
             length = getLengthFromJsMap(sample.getMap("length")),
           )
         } ?: emptyList(),
-        segments = it.getArray("samples")?.toMapList()?.map { sample ->
+        segments = it.getArray("segments")?.toMapList()?.map { sample ->
           ExerciseSegment(
             startTime = Instant.parse(sample.getString("startTime")),
             endTime = Instant.parse(sample.getString("endTime")),
